### ISSUE TITLE
Typescript not working 

### ISSUE THREAD

**Issue by vforv** [hwchase17/langchainjs#340]
> ts-node -T --logError test-init.ts

```
error: uncaughtException No "exports" main defined i/node_modules/langchain/package.json {"timestamp":"March-15-2023 18:43:49"}
error: uncaughtException Stack Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in node_modules/langchain/package.json
    at new NodeError (node:internal/errors:387:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:365:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:589:7)
    at resolveExports (node:internal/modules/cjs/loader:529:36)
    at Function.Module._findPath (node:internal/modules/cjs/loader:569:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:981:2
```

I tried also deleting package.json but it just create new error

**Comment by ayalpani** [comment]
Mine looks like 

`Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './chains' is not defined by "exports" in /Users/ayalpani/dev/playground/node_modules/langchain/package.json`

**Comment by abhishek-codaio** [comment]
> Mine looks like
> 
> `Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './chains' is not defined by "exports" in /Users/ayalpani/dev/playground/node_modules/langchain/package.json`

I'm also seeing this: `Error: Package subpath './llms' is not defined by "exports" in <PATH>/node_modules/langchain/package.json`

**Comment by nsbradford** [comment]
Looks like https://github.com/hwchase17/langchainjs/issues/185 (response there)

**Comment by indiejoseph** [comment]
still not working with Typescript, even added { "type": "module" } in packages.json, and running on NodeJS v8

**Comment by AuriDeimantasDev** [comment]
Inline imports should fix this issue. Fx:

Before
```typescript
import {VectorDBQAChain} from 'langchain/chains';
```

After
```typescript
const {VectorDBQAChain} = await import('langchain/chains');
```

**Comment by JimTeva** [comment]
> > Mine looks like
> > `Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './chains' is not defined by "exports" in /Users/ayalpani/dev/playground/node_modules/langchain/package.json`
> 
> I'm also seeing this: `Error: Package subpath './llms' is not defined by "exports" in <PATH>/node_modules/langchain/package.json`

I got the same error then checked again requirements for typescript [here](https://js.langchain.com/docs/getting-started/install#typescript) and it says:
```
{
  "compilerOptions": {
    ...
    "target": "ES2020", // or higher
    "module": "nodenext",
  }
}
````
Updated my tsconfig.json then it worked

**Comment by Marviel** [comment]
have performed both fixes mentioned above and still no dice, receiving:
```
Error: Must use import to load ES Module: /Users/lukebechtel/localdev/lib/apps/scripts/src/agents/simple-notion-agent.ts
```

**Comment by nfcampos** [comment]
Hi, we've released a new version 0.0.49 which adds a CJS build and should fix this issue.

**Comment by hilarykitz** [comment]
If anyone is still stuck on this, make sure that you are explicitly stating  `"type": "commonjs"` in package.json and `"module": "CommonJS"` in your tsconfig :)
